core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestUseConnectorTableScanPartitioning.java

```diff
@@ -0,0 +1,308 @@
+                false);
+    }
+
+    @Test
+    public void testTableScanWithFixedConnectorPartitioning()
+    {
+        testPlanWithTableNodePartitioning(
+                DISABLE_PLAN_WITH_TABLE_NODE_PARTITIONING,
+                FIXED_PARTITIONED_TABLE_HANDLE,
+                CONNECTOR_FIXED_PARTITIONED_TABLE_HANDLE,
+                true);
+    }
+
+    private void testPlanWithTableNodePartitioning(
+            Session session,
+            TableHandle tableHandle,
+            ConnectorTableHandle connectorTableHandle,
+            boolean expectedEnabled)
+    {
+        tester.assertThat(new UseConnectorTableScanPartitioning(tester.getMetadata(), tester.getQueryRunner().getNodePartitioningManager()))
+                .on(p -> {
+                    Symbol a = p.symbol(COLUMN_A);
+                    Symbol b = p.symbol(COLUMN_B);
+                    return p.tableScan(tableHandle,
+                            ImmutableList.of(a, b),
+                            ImmutableMap.of(a, COLUMN_HANDLE_A, b, COLUMN_HANDLE_B));
+                })
+                .withSession(session)
+                .matches(
+                        tableScan(
+                                equalTo(connectorTableHandle),
+                                TupleDomain.all(),
+                                ImmutableMap.of(
+                                        "A", equalTo(COLUMN_HANDLE_A),
+                                        "B", equalTo(COLUMN_HANDLE_B)))
+                                .with(planWithTableNodePartitioning(expectedEnabled)));
+    }
+
+    private Matcher planWithTableNodePartitioning(boolean enabled)
+    {
+        return new Matcher()
+        {
+            @Override
+            public boolean shapeMatches(PlanNode node)
+            {
+                return node instanceof TableScanNode;
+            }
+
+            @Override
+            public MatchResult detailMatches(PlanNode node, StatsProvider stats, Session session, Metadata metadata, SymbolAliases symbolAliases)
+            {
+                TableScanNode tableScanNode = (TableScanNode) node;
+                if (tableScanNode.getUseConnectorTablePartitioning().isEmpty()) {
+                    return NO_MATCH;
+                }
+
+                if (tableScanNode.isUseConnectorTablePartitioning() != enabled) {
+                    return NO_MATCH;
+                }
+
+                return MatchResult.match();
+            }
+        };
+    }
+
+    public static MockConnectorFactory createMockFactory()
+    {
+        MockConnectorFactory.Builder builder = MockConnectorFactory.builder();
+        builder
+                .withGetColumns(schemaTableName -> ImmutableList.of(
+                        new ColumnMetadata(COLUMN_A, BIGINT),
+                        new ColumnMetadata(COLUMN_B, VARCHAR)))
+                .withGetTableProperties((session, tableHandle) -> {
+                    if (tableHandle.equals(CONNECTOR_PARTITIONED_TABLE_HANDLE)) {
```

Comment:
switch?

Reply:
unfortunately swtich doesn' work with objects in Java 11